smarts/core/chassis.py

```diff
@@ -509,9 +515,11 @@ def step(self, current_simulation_time):
 
     def inherit_physical_values(self, other: BoxChassis):
         self.set_pose(other.pose)
-        # bullet wants m/s
-        velocity = radians_to_vec(self.pose.heading) * other.speed
-        self.speed = other.speed
+        self.reset_speed(other.speed)
+
+    def reset_speed(self, speed: float):
```

Comment:
Is `set_speed` a better naming? And since the chassis already has `speed.setter`, why adding a `reset_speed`?

Reply:
In this case the reason is that this method should only be called before the chassis sees any operation on it.

Reply:
I ended up removing it.